tabs/auxiliary.js

```diff
@@ -271,12 +276,53 @@ TABS.auxiliary.initialize = function (callback) {
                 }
             }
 
+            auto_select_channel(RC.channels);
+
             var auxChannelCount = RC.active_channels - 4;
 
             for (var i = 0; i < (auxChannelCount); i++) {
                 box_highlight(i, RC.channels[i + 4]);
                 update_marker(i, RC.channels[i + 4]);
-            }           
+            }
+        }
+
+        /**
+         * Autodetect channel based on maximum deference with previous value
+         * minimum value to autodetect is 100 - to prevent auto select RSSI channel
+         * @param RC_channels
+         */
+        function auto_select_channel(RC_channels) {
+            var auto_option = $('.tab-auxiliary select.channel option[value="-1"]:selected');
+            if (auto_option.length === 0) {
+                prevChannelsValues = null;
+                return;
+            }
+
+            if (!prevChannelsValues) {
+                prevChannelsValues = RC_channels.slice(0); //clone array
+                return;
+            }
+
+            var diff_array = RC_channels.map(function(currentValue, index) {
+                return Math.abs(prevChannelsValues[index] - currentValue);
+            });
+
+            var largest = diff_array.reduce(function(x,y){
```

Comment:
This will throw an exception if no channels are available (i.e. no RX is configured), since `diff_array` is empty in this case:

```
Error in event handler for serial.onReceive: TypeError: Reduce of empty array with no initial value
    at auto_select_channel (chrome-extension://bghjhhpmeamhngbeonnholbimdejoghm/tabs/auxiliary.js:310:38)
    at update_ui (chrome-extension://bghjhhpmeamhngbeonnholbimdejoghm/tabs/auxiliary.js:279:13)
    at MspHelper.process_data (chrome-extension://bghjhhpmeamhngbeonnholbimdejoghm/js/msp/MSPHelper.js:1182:31)
    at chrome-extension://bghjhhpmeamhngbeonnholbimdejoghm/js/msp.js:139:13
    at Object.notify (chrome-extension://bghjhhpmeamhngbeonnholbimdejoghm/js/msp.js:138:24)
    at Object.read (chrome-extension://bghjhhpmeamhngbeonnholbimdejoghm/js/msp.js:126:26)
    at read_serial (chrome-extension://bghjhhpmeamhngbeonnholbimdejoghm/js/serial_backend.js:397:13)
```

Reply:
This is still throwing an exception...

Reply:
Now should be good

Reply:
Nope, still broken. The problem seems to be that `diff_array` is an array of 32 empty elements (`(32) [empty × 32]length: 32__proto__: Array(0)`) in the case that no RX is configured.

Reply:
Sorry, now I get in what was problem